plugins/search/src/alpha.tsx

```diff
@@ -0,0 +1,485 @@
+                      className={classes.filter}
+                      label="Kind"
+                      name="kind"
+                      values={['Component', 'Template']}
+                    />
+                    <SearchFilter.Checkbox
+                      className={classes.filter}
+                      label="Lifecycle"
+                      name="lifecycle"
+                      values={['experimental', 'production']}
+                    />
+                  </Paper>
+                </Grid>
+              )}
+              <Grid item xs>
+                <SearchPagination />
+                <SearchResults>
+                  {({ results }) =>
+                    results.map((result, index) => {
+                      const SearchResultListItem =
+                        getResultItemComponent(result);
+                      return (
+                        <SearchResultListItem
+                          key={index}
+                          rank={result.rank}
+                          result={result.document}
+                          noTrack
+                        />
+                      );
+                    })
+                  }
+                </SearchResults>
+                <SearchResultPager />
+              </Grid>
+            </Grid>
+          </Content>
+        </Page>
+      );
+    };
+
+    return (
+      <SearchContextProvider>
+        <UrlUpdater />
+        <Component />
+      </SearchContextProvider>
+    );
+  },
+});
+
+const searchModalExtensionData = createExtensionDataRef<() => JSX.Element>(
+  'plugin.search.modal.element',
+);
+
+const useSearchModalStyles = makeStyles(theme => ({
+  dialogTitle: {
+    gap: theme.spacing(1),
+    display: 'grid',
+    alignItems: 'center',
+    gridTemplateColumns: '1fr auto',
+    '&> button': {
+      marginTop: theme.spacing(1),
+    },
+  },
+  container: {
+    borderRadius: 30,
+    display: 'flex',
+    height: '2.4em',
+    padding: theme.spacing(1),
+  },
+  filter: {
+    '& + &': {
+      marginTop: theme.spacing(2.5),
+    },
+  },
+  filters: {
+    padding: theme.spacing(2),
+    marginTop: theme.spacing(2),
+  },
+  input: {
+    flex: 1,
+  },
+  button: {
+    '&:hover': {
+      background: 'none',
+    },
+  },
+  dialogActionsContainer: { padding: theme.spacing(1, 3) },
+  viewResultsLink: { verticalAlign: '0.5em' },
+}));
+
+/** @alpha */
+export const SearchModal = createExtension({
+  id: 'plugin.search.modal',
+  at: 'plugin.search.nav.index/modal',
+  inputs: {
+    items: createExtensionInput({
+      item: searchResultItemExtensionData,
+    }),
+  },
+  output: {
+    element: searchModalExtensionData,
+  },
+  factory({ bind, inputs }) {
+    const getResultItemComponent = (result: SearchResult) => {
+      const value = inputs.items.find(({ item }) => item?.predicate?.(result));
+      return value?.item.component ?? DefaultResultListItem;
+    };
+
+    const Component = () => {
```

Comment:
> [!Note]
This is the default `SearchModal` which was previously configured via code on the non-declarative integrated App ([packages/app/src/components/search/SearchPage](https://github.com/backstage/backstage/blob/master/packages/app/src/components/search/SearchModal.tsx)).

Reply:
Removed.